zsh test failures on Fedora

Hello,

I'm working on an update to the argcomplete Fedora package from 2.0.0 to 3.1.1. When running the tests with python3 test/test.py I'm getting the following errors with [zsh-5.9](https://koji.fedoraproject.org/koji/buildinfo?buildID=2276858):

<details>
<summary>Errors</summary>
======================================================================
FAIL: test_comp_point (__main__.TestZsh.test_comp_point)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_completion_environment (__main__.TestZsh.test_completion_environment)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_continuation (__main__.TestZsh.test_continuation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_double_quoted_completion (__main__.TestZsh.test_double_quoted_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_exclamation_in_double_quotes (__main__.TestZsh.test_exclamation_in_double_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_parse_special_characters (__main__.TestZsh.test_parse_special_characters)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_parse_special_characters_dollar (__main__.TestZsh.test_parse_special_characters_dollar)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_partial_completion (__main__.TestZsh.test_partial_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_quoted_exact (__main__.TestZsh.test_quoted_exact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_quoted_space (__main__.TestZsh.test_quoted_space)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_quotes (__main__.TestZsh.test_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_simple_completion (__main__.TestZsh.test_simple_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_single_quoted_completion (__main__.TestZsh.test_single_quoted_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_single_quotes_in_double_quotes (__main__.TestZsh.test_single_quotes_in_double_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_single_quotes_in_single_quotes (__main__.TestZsh.test_single_quotes_in_single_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_special_characters (__main__.TestZsh.test_special_characters)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_special_characters_double_quoted (__main__.TestZsh.test_special_characters_double_quoted)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_special_characters_single_quoted (__main__.TestZsh.test_special_characters_single_quoted)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_unquoted_space (__main__.TestZsh.test_unquoted_space)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
======================================================================
FAIL: test_wordbreak_chars (__main__.TestZsh.test_wordbreak_chars)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.1/test/test.py", line 1255, in setUp
    self.assertEqual(output, "")
AssertionError: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\[807 chars]\x00' != ''
- ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
----------------------------------------------------------------------
</details>

Thank you for taking a look at this. With the patch applied the output looks as follows.

<details><summary>test output</summary>
<p>

```python
======================================================================
FAIL: test_double_quoted_completion (__main__.TestZsh.test_double_quoted_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_exclamation_in_double_quotes (__main__.TestZsh.test_exclamation_in_double_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_partial_completion (__main__.TestZsh.test_partial_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_quoted_exact (__main__.TestZsh.test_quoted_exact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_quoted_space (__main__.TestZsh.test_quoted_space)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_quotes (__main__.TestZsh.test_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_simple_completion (__main__.TestZsh.test_simple_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_single_quoted_completion (__main__.TestZsh.test_single_quoted_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_single_quotes_in_double_quotes (__main__.TestZsh.test_single_quotes_in_double_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_single_quotes_in_single_quotes (__main__.TestZsh.test_single_quotes_in_single_quotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_special_characters (__main__.TestZsh.test_special_characters)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_special_characters_double_quoted (__main__.TestZsh.test_special_characters_double_quoted)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_special_characters_single_quoted (__main__.TestZsh.test_special_characters_single_quoted)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


======================================================================
FAIL: test_unquoted_space (__main__.TestZsh.test_unquoted_space)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-3.1.6/test/test.py", line 1255, in setUp
    self.assertEqual(output, "ready\r\n")
AssertionError: 'ready\r\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' != 'ready\r\n'
  ready
- 
+ 


----------------------------------------------------------------------
Ran 127 tests in 64.260s

FAILED (failures=14, skipped=6, expected failures=5)
```

</p>
</details>

--- a/test/test.py
+++ b/test/test.py
@@ -54,7 +54,11 @@ class ArgcompleteREPLWrapper(REPLWrapper):
                 raise Exception("Expected to see a newline in command response")
             echo_cmd, actual_res = res.split("\n", 1)
             res_without_ansi_seqs = re.sub(r"\x1b\[0m.+\x1b\[J", "", actual_res)
-            return res_without_ansi_seqs
+            # Unsure why some environments produce trailing null characters,
+            # but they break tests and trimming them seems to be harmless.
+            # https://github.com/kislyuk/argcomplete/issues/447
+            res_without_null_chars = res_without_ansi_seqs.rstrip("\x00")
+            return res_without_null_chars
         else:
             return res
 

